Jump to errors section V1 ; WIP

sravfeyn opened the pull request:
## Summary

Implements a UI to jump to errored questions in Webapps. Clicking on a question scrolls the UI to that question.
<img width="1123" alt="Screenshot 2021-04-06 at 2 49 38 PM" src="https://user-images.githubusercontent.com/829142/113688919-61f52680-96e7-11eb-90c6-eef1e71b6be0.png">



## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

TBD

### Safety story
<!--
Describe any other pieces to the safety story including
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.
-->

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

sravfeyn committed:
Jump to errors section V1

sravfeyn committed:
Alert

sravfeyn committed:
lint

sravfeyn committed:
error class

sravfeyn committed:
Move error box, disable button on errors

sravfeyn committed:
Scroll to bottom button

sravfeyn committed:
handle collapsible groups

sravfeyn committed:
Make jumpToErrors button

sravfeyn committed:
correct message

sravfeyn committed:
handle nested groups

sravfeyn committed:
flash div

sravfeyn committed:
better flow

sravfeyn committed:
product feedback

sravfeyn committed:
change colors

sravfeyn committed:
Product notification

sravfeyn committed:
lint

orangejenny commented on corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js:
```diff
@@ -542,6 +592,29 @@ hqDefine("cloudcare/js/form_entry/fullform-ui", function () {
+            $('html, body').animate({
+                scrollTop: $(el).offset().top - 60,
+            });
+            el.fadeOut(200).fadeIn(200).fadeOut(200).fadeIn(200);
```
What does this do? Make the label for the errored question fade in & out to catch the user's attention?

sravfeyn replied:
It blinks the question, useful when there is only one required question and clicking the button repeatedly gives this feedback without which the button feels like doing nothing

orangejenny replied:
cool!

orangejenny committed:
Moved notification to top of page

orangejenny committed:
Updated notification text

orangejenny committed:
Updated error message to render HTML from questions

orangejenny committed:
Updated notification to not show in live preview

orangejenny committed:
Fixed operator

orangejenny committed:
Removed new UI from app preview

orangejenny committed:
Moved notification to point to button

orangejenny committed:
Added logic to prevent dismissed alert from re-appearing on future forms

orangejenny committed:
Updated size of notification

orangejenny committed:
Fixed client-side errors, like data type validation, not displaying.

Also removed server-error-message class, which wasn't doing anything,
and removed the bolding from the new UI.

orangejenny committed:
Fixed lint errors

orangejenny committed:
Switched notification condition

This hides the notification from tests, which are erroring on Travis only.

orangejenny committed:
Added "Unknown Question"
